[KOGITO-3067] - Migrate to Operator SDK 1.x

Kaitou786 opened the pull request:
See: https://issues.redhat.com/browse/KOGITO-3067

Signed-off-by: Tarun Khandelwal <tarkhand@redhat.com>

Many thanks for submitting your Pull Request :heart:! 

Please make sure your PR meets the following requirements:

- [x] You have read the [contributors' guide](https://github.com/kiegroup/kogito-cloud-operator/blob/master/README.md#contributing-to-the-kogito-operator)
- [x] Pull Request title is properly formatted: `[KOGITO-XYZ] Subject`
- [x] Pull Request contains a link to the JIRA issue
- [x] Pull Request contains a description of the issue
- [x] Pull Request does not include fixes for issues other than the main ticket
- [x] Your feature/bug fix has a unit test that verifies it
- [x] You've tested the new feature/bug fix in an actual OpenShift cluster
- [x] You've added a [RELEASE_NOTES.md](RELEASE_NOTES.md) entry regarding this change

Kaitou786 commented:
Don't run the BDD test till now, the image would need to be updated with 1.2.0 operator-sdk and I believe the test would also need some updates :p

sutaakar commented:
@Kaitou786 Please update Requirements section in readme.

ricardozanini reviewed (changes requested):
Great work man! I believe we have to do small adjustments only. This is a huge step for us! Many thanks!

sutaakar commented on Makefile:
```diff
@@ -1,60 +1,147 @@
+
+# Build the docker image
+IMAGE ?= $(IMG)
+BUILDER ?= docker
```
```suggestion
BUILDER ?= podman
```

sutaakar commented on Makefile:
```diff
@@ -1,60 +1,147 @@
+
+# Push the docker image
+docker-push:
+	docker push ${IMG}
```
```suggestion
	$(BUILDER) push ${IMG}
```

sutaakar commented on Makefile:
```diff
@@ -1,60 +1,147 @@
+# Build the bundle image.
+.PHONY: bundle-build
+bundle-build:
+	docker build -f bundle.Dockerfile -t $(BUNDLE_IMG) .
```
```suggestion
	$(BUILDER) build -f bundle.Dockerfile -t $(BUNDLE_IMG) .
```

sutaakar commented:
@Kaitou786 I tried to build the operator image and deploy it. But the operator pod doesn't start with this error:
```
Error: container create failed: time="2020-12-14T14:12:43Z" level=error msg="container_linux.go:366: starting container process caused: exec: \"/manager\": stat /manager: no such file or directory" 
```
Do you have any idea what can be wrong? Installed using `oc apply -f kogito-operator.yaml`

ricardozanini commented:
@sutaakar I'm assuming you upgraded your `operator-sdk` binaries locally?

sutaakar commented:
@ricardozanini Yes, will try to cleanup CRC first, maybe there was cached old image. When I run the image manually locally the operator seems to be started.

ricardozanini commented:
> @ricardozanini Yes, will try to cleanup CRC first, maybe there was cached old image. When I run the image manually locally the operator seems to be started.

I'd say you could try running the image first locally. This is an entrypoint error, so you would see a k8s connection error instead.

sutaakar commented:
It is running now on clean CRC.

Kaitou786 commented on bundle/manifests/kogito-operator.clusterserviceversion.yaml:
```diff
@@ -0,0 +1,340 @@
+  - supported: true
+    type: MultiNamespace
+  - supported: true
+    type: AllNamespaces
```
Yes, we'll support everything, although in OLM user only has two options either a specific namespace or all namespaces

Kaitou786 commented on cmd/kogito/command/service/build_service.go:
```diff
@@ -124,14 +123,13 @@ func (i buildService) InstallBuildService(flags *flag.BuildFlags, resource strin
 }
 
-func (i buildService) validatePreRequisite(flags *flag.BuildFlags, log *zap.SugaredLogger) error {
+func (i buildService) validatePreRequisite(flags *flag.BuildFlags, log context.Logger) error {
```
I believe this will be handled in the refactoring that will be done by @vaibhavjainwiz for our core and services package

Kaitou786 commented on cmd/kogito/command/context/logger.go:
```diff
@@ -15,25 +15,59 @@
 package context
 
 import (
-	"github.com/kiegroup/kogito-cloud-operator/pkg/logger"
```
Created [KOGITO-4017](https://issues.redhat.com/browse/KOGITO-4017)

Kaitou786 commented on .github/workflows/Kogito-Operator-PR-Check.yaml:
```diff
@@ -34,29 +35,29 @@ jobs:
+          make vet
           git diff
-          changed_files=$(git status -s | grep -v 'go.mod\|go.sum' || :)
+          changed_files=$(git status -s | grep -v 'go.mod\|go.sum\|kogito-operator.yaml' || :)
```
Not sure why but the indentation of the description changes on this GH machine, [see](https://github.com/kiegroup/kogito-cloud-operator/pull/695/checks?check_run_id=1562481995#step:7:72) I tried changing it like this locally but running `make vet` reverts it. Although if someone forgot to run `make vet` we'll get know through change in other auto generated files,

sutaakar commented on Makefile:
```diff
@@ -1,60 +1,141 @@
-ifeq ("$(origin V)", "command line")
-       BUILD_VERBOSE = $(V)
+# Current Operator version
+VERSION ?= 2.0.0-snapshot
```
This duplicates version information placed in `pkg/version/version.go`. Can we retrieve version information from there? Or at least cover update of this line in `bump-version.sh`?

Kaitou786 replied:
We use the version written in `pkg/version/version.go` to form the default image tag as well. The bump-version.sh script will update this line. [See](https://github.com/Kaitou786/kogito-cloud-operator/blob/KOGITO-3067/hack/bump-version.sh#L41)

sutaakar replied:
Thanks, I overlooked that

sutaakar commented on Makefile:
```diff
@@ -1,60 +1,141 @@
+
+# Push the docker image
+docker-push:
+	$(BUILDER) push ${IMG}
```
Why docker build use parameter `IMAGE` but docker push use `IMG`

Kaitou786 replied:
True, will change everywhere to use `IMAGE` only

sutaakar commented on PROJECT:
```diff
@@ -0,0 +1,20 @@
+domain: kiegroup.org
```
What is the purpose of this file?

Kaitou786 replied:
This the file created and used by the operator-sdk cli to create CSV and validate the bundle, it has meta information like our project name and the API we expose

sutaakar replied:
Ok, do we need to persist it on GitHub? If it is generated by operator-sdk then we can IMHO exclude it in gitignore.

Kaitou786 replied:
This file is not generated everytime, only once when we scaffold the project. And without it all CSV and buncle creation command will fail with `FATA[0000] error reading configuration: open PROJECT: no such file or directory`

sutaakar replied:
aha, ok, thanks for the context :)

sutaakar commented:
@Kaitou786 When I deploy the operator using `make deploy` then I can see this error:
```
2020-12-16T13:14:38.791Z	ERROR	controller	Could not wait for Cache to sync	{"controller": "KogitoSupportingService-finalizer", "error": "failed to wait for KogitoSupportingService-finalizer caches to sync: cache did not sync"}
github.com/go-logr/zapr.(*zapLogger).Error
	/go/pkg/mod/github.com/go-logr/zapr@v0.1.1/zapr.go:128
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func1
	/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.6.3/pkg/internal/controller/controller.go:160
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start
	/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.6.3/pkg/internal/controller/controller.go:184
sigs.k8s.io/controller-runtime/pkg/manager.(*controllerManager).startRunnable.func1
	/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.6.3/pkg/manager/internal.go:661
```
Do you have any idea what can be wrong?

Kaitou786 commented:
Were the CRD installed already in the cluster?

sutaakar commented:
From the bash logs I would expect the CRDs to be installed too:
```
[karel@localhost kogito-cloud-operator]$ make deploy
/home/karel/go/bin/controller-gen "crd:trivialVersions=true" rbac:roleName=manager-role webhook paths="./..." output:crd:artifacts:config=config/crd/bases
cd config/manager && /home/karel/go/bin/kustomize edit set image controller=quay.io/kiegroup/kogito-cloud-operator:2.0.0-snapshot
/home/karel/go/bin/kustomize build config/default | kubectl apply -f -
namespace/kogito-operator-system created
Warning: apiextensions.k8s.io/v1beta1 CustomResourceDefinition is deprecated in v1.16+, unavailable in v1.22+; use apiextensions.k8s.io/v1 CustomResourceDefinition
customresourcedefinition.apiextensions.k8s.io/kogitobuilds.app.kiegroup.org configured
customresourcedefinition.apiextensions.k8s.io/kogitoinfras.app.kiegroup.org configured
customresourcedefinition.apiextensions.k8s.io/kogitoruntimes.app.kiegroup.org configured
customresourcedefinition.apiextensions.k8s.io/kogitosupportingservices.app.kiegroup.org configured
role.rbac.authorization.k8s.io/kogito-operator-leader-election-role created
clusterrole.rbac.authorization.k8s.io/kogito-operator-manager-role unchanged
clusterrole.rbac.authorization.k8s.io/kogito-operator-metrics-reader unchanged
clusterrole.rbac.authorization.k8s.io/kogito-operator-proxy-role unchanged
rolebinding.rbac.authorization.k8s.io/kogito-operator-leader-election-rolebinding created
clusterrolebinding.rbac.authorization.k8s.io/kogito-operator-manager-rolebinding unchanged
clusterrolebinding.rbac.authorization.k8s.io/kogito-operator-proxy-rolebinding unchanged
service/kogito-operator-controller-manager-metrics-service created
deployment.apps/kogito-operator-controller-manager created
```

sutaakar commented:
Also I see `DEBUG` entries in operator container log (there are no env variables defined), is it expected?

Kaitou786 commented:
> Also I see `DEBUG` entries in operator container log (there are no env variables defined), is it expected?

Yes, it is expected, I am trying to debug the above the error

ricardozanini reviewed (approved):
Looks good overall! I'm sure I'm letting something off, but that's life. 479 files to review is not as easy as it looks. Hopefully, @sutaakar will help us verify this work. The quicker we get this merged, the better to test with actual code. Also, @vaibhavjainwiz is working on the package restructure, he might get something in there as well.

sutaakar commented on cmd/kogito/command/deploy/delete_service_test.go:
```diff
@@ -35,7 +35,7 @@ func Test_DeleteServiceCmd_SuccessfullyDelete(t *testing.T) {
 	lines, _, err := test.ExecuteCli()
 	assert.NoError(t, err)
-	assert.Contains(t, lines, "Successfully deleted Kogito Service example-drools")
+	assert.Contains(t, lines, "Successfully deleted")
```
I have noticed that some log entriesin CLI are adjusted, for example:
```
[karel@localhost kogito-cloud-operator]$ kogito new-project ksuta-test
Successfully created	{"Project": "ksuta-test"}
[karel@localhost kogito-cloud-operator]$ kogito deploy-service test
Creating resource	{"kind": "", "name": "test", "namespace": "ksuta-test"}
Kogito Build Service successfully installed in the Project ksuta-test.
Check the Service status by running 'oc describe kogitobuild/test -n ksuta-test'
Your Kogito Runtime Service needs the application binaries to proceed. To upload your binaries please run 'oc start-build test --from-dir=target -n ksuta-test' from your project's root
Creating resource	{"kind": "", "name": "test", "namespace": "ksuta-test"}
Kogito Service successfully installed in the Project ksuta-test.
Check the Service status by running 'oc describe kogitoruntime/test -n ksuta-test'
To more easily manage your Kogito Service install Data Index Service and Process Instance Management. 
For how to install see: https://docs.jboss.org/kogito/release/latest/html_single/#con-kogito-operator-with-data-index-service_kogito-deploying-on-openshift
and https://docs.jboss.org/kogito/release/latest/html_single/#con-management-console_kogito-developing-process-services
```
I found it a bit hard to understand what actually happened with this JSON format logs, the previous approach with natural language was more readable. Is there a specific reason for using such JSON format? Or is it a preferred approach now?

Kaitou786 replied:
This new format kind of JSON is the new preferred approach and is the only way to define logs with the logger implementation comes with the operator-sdk. We need to define everything in `"key", "value"` so formatted logs are not allowed, There is a trick for using the formatted logs that is used to print logs like this one 
```
Your Kogito Runtime Service needs the application binaries to proceed. To upload your binaries please run 'oc start-build test --from-dir=target -n ksuta-test' from your project's root
``` since it requires some addition in the middle but it is not recommended. For more about [this logger see](https://github.com/go-logr/logr)

sutaakar replied:
hmm, ok

ricardozanini replied:
@Kaitou786 I don't think this JSON approach is a good solution for the CLI. I'd rather get back to the old format for the CLI. In this case we can drop the `klog` module to use `loggrus` instead. The last one is used by minikube and CRC as well.

Kaitou786 replied:
Seeing loggrus is also a good option but that too doesn't fullly support the formatted(human friendly logging) log that we would want to use with our CLI, [see](https://github.com/sirupsen/logrus#level-logging). Currently we switched to our earlier zap logger for the formatted logging for not changing too much.

ricardozanini replied:
> Seeing loggrus is also a good option but that too doesn't fullly support the formatted(human friendly logging) log that we would want to use with our CLI, [see](https://github.com/sirupsen/logrus#level-logging). 

I'm not sure if I follow. Why is it not a good option for human-friendly logging? Using loggrus I can see an output like:

```
INFO Checking if podman remote executable is cached 
INFO Checking if goodhosts executable is cached   
INFO Checking minimum RAM requirements            
INFO Checking if running as non-root              
INFO Checking if Virtualization is enabled        
INFO Checking if KVM is enabled                   
INFO Checking if libvirt is installed             
INFO Checking if user is part of libvirt group    
INFO Checking if libvirt daemon is running 
```

Doesn't this help? Also, it's used by crc and operator-sdk. We can relate and even use with BDD since they can output to files as well.

But seeing this disclaimer:

> I believe Logrus' biggest contribution is to have played a part in today's widespread use of structured logging in Golang. There doesn't seem to be a reason to do a major, breaking iteration into Logrus V2, since the fantastic Go community has built those independently. Many fantastic alternatives have sprung up. Logrus would look like those, had it been re-designed with what we know about structured logging in Go today. Check out, for example, Zerolog, Zap, and Apex.

I believe that we should stick to Zap.

Kaitou786 commented:
> @Kaitou786 When I deploy the operator using `make deploy` then I can see this error:
> 
> ```
> 2020-12-16T13:14:38.791Z	ERROR	controller	Could not wait for Cache to sync	{"controller": "KogitoSupportingService-finalizer", "error": "failed to wait for KogitoSupportingService-finalizer caches to sync: cache did not sync"}
> github.com/go-logr/zapr.(*zapLogger).Error
> 	/go/pkg/mod/github.com/go-logr/zapr@v0.1.1/zapr.go:128
> sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func1
> 	/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.6.3/pkg/internal/controller/controller.go:160
> sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start
> 	/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.6.3/pkg/internal/controller/controller.go:184
> sigs.k8s.io/controller-runtime/pkg/manager.(*controllerManager).startRunnable.func1
> 	/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.6.3/pkg/manager/internal.go:661
> ```
> 
> Do you have any idea what can be wrong?

The error should be fixed with the latest changes, thanks :)

sutaakar commented on cmd/kogito/command/remove/supportingservices.go:
```diff
@@ -135,14 +135,14 @@ func (r *removeSupportingServiceCommand) Exec(cmd *cobra.Command, args []string)
 
 	if len(targetServiceItems) == 0 {
-		log.Warnf("There's no service %s in the namespace %s", r.supportingService.cmdName, r.flags.namespace)
+		log.Warn("There's no", "service", r.supportingService.cmdName, "namespace", r.flags.namespace)
```
Would be good to add a service type to the lo table.

ricardozanini reviewed (changes requested):
See my comment regarding logging in the CLI.

sutaakar commented on config/rbac/auth_proxy_service.yaml:
```diff
@@ -0,0 +1,14 @@
+apiVersion: v1
+kind: Service
```
Optional nitpick: the service could be placed somewhere else than in rbac. Maybe in the default folder?

Kaitou786 replied:
Not sure if this should be done. All the files in the config dir is generated by the operator-sdk and I believe they expect in the subcommands that dir structure would be same to the one they create with different content (if required)

sutaakar replied:
Aha, ok. I didn't know that this was generated.

sutaakar commented on config/rbac/role_binding.yaml:
```diff
@@ -0,0 +1,12 @@
+  name: manager-role
+subjects:
+- kind: ServiceAccount
+  name: default
```
So we are using default ServiceAccount for Kogito operator now? Wouldn't it cause some potential security risk in case of namespace scoped deployments?

Kaitou786 replied:
I am not sure but we would have to test how  and which SA OLM would use. IIRC in the CSV we don't define role_binding definition, just the permission and cluster permission required by the operator. This should only affect the yaml install and I believe we should push the OLM installation to users :). For the OLM I will deploy and see which SA is created and used :)

Kaitou786 replied:
I'll create a service_account.yaml with name as `kogito-operator` and give its reference in kustomization.yaml and role_binding.yaml . By default OLM gives the permstion default service account which as you mentioned is not good for security reasons. Thanks for pointing this :)

ricardozanini replied:
This is not a problem as I see it. This account is created and managed within the namespace the operator is installed.

Kaitou786 replied:
So we keep it as it is with default SA?

ricardozanini replied:
Yes

Kaitou786 replied:
Okay

sutaakar commented on controllers/suite_test.go:
```diff
@@ -0,0 +1,104 @@
+/*
+
+
+Licensed under the Apache License, Version 2.0 (the "License");
```
Do we need two licence headers here?

sutaakar replied:
Was this file autogenerated too?

Kaitou786 replied:
This was autogenerated but I do belive we should remove the extra license and only keep the Red Hat one that we have on other files, there might be some other files as well with this kind of two license I'll remove the extra one from them as well :)

sutaakar replied:
Thanks, I have noticed just this one occurrence.
Btw do we need this file at all? It looks like it is used for Ginkgo tests which AFAIK we don't use.

ricardozanini replied:
These tests are a nice way to test the controllers. We will need in the long run.

sutaakar commented on main.go:
```diff
@@ -0,0 +1,151 @@
+		Log:    logger.GetLogger("KogitoInfra-controller"),
+		Scheme: mgr.GetScheme(),
+	}).SetupWithManager(mgr); err != nil {
+		setupLog.Error(err, "unable to create controller", "controller", "KogitoInfra")
```
```suggestion
		setupLog.Error(err, "unable to create controller", "controller", "KogitoInfra - finalizer")
```

sutaakar commented on main.go:
```diff
@@ -0,0 +1,151 @@
+		Log:    logger.GetLogger("KogitoInfra-controller"),
+		Scheme: mgr.GetScheme(),
+	}).SetupWithManager(mgr); err != nil {
+		setupLog.Error(err, "unable to create controller", "controller", "KogitoInfra")
```
```suggestion
		setupLog.Error(err, "unable to create controller", "controller", "KogitoRuntime - finalizer")
```

sutaakar commented on test/features/deploy_kogito_runtime.feature:
```diff
@@ -351,18 +351,18 @@ Feature: Deploy Kogito Runtime
     Then Service "process-mongodb" contains 1 instances of process with name "dealreviews" within 2 minutes
 
-    @springboot
+  @springboot
```
The previous annotation alignment was correct one, can you please return it back?

sutaakar reviewed (commented):
Some minor points above, otherwise the code looks good.
Great job :)

Still need to try some manual deployments and checks with operator image. Also need to find out how to adjust BDD tests for this.

radtriste commented on hack/export-version.sh:
```diff
@@ -14,6 +14,13 @@
+
+### Fetching the latest released kogito-cloud-operator on the OperatorHub
+rm -rf /tmp/operators/
+git clone https://github.com/operator-framework/community-operators/ /tmp/operators/
+LATEST_RELEASED_OLM_VERSION=$(cd /tmp/operators/community-operators/kogito-operator && for i in $(ls -d */); do echo ${i%%/}; done | sort -V | tail -1)
```
I like this new approach :)

radtriste commented on hack/go-test.sh:
```diff
@@ -13,23 +13,15 @@
+  ENVTEST_ASSETS_DIR=testbin
 fi
+
+mkdir -p $(pwd)/${ENVTEST_ASSETS_DIR}
```
maybe store the result of `$(pwd)/${ENVTEST_ASSETS_DIR}` in a var to be reused after instead of repeating the command each time

Kaitou786 replied:
Changed thanks :)

radtriste reviewed (commented):
Sorry for the late review ...

Kaitou786 commented on Makefile:
```diff
@@ -1,60 +1,141 @@
+BUNDLE_METADATA_OPTS ?= $(BUNDLE_CHANNELS) $(BUNDLE_DEFAULT_CHANNEL)
+
+# Image URL to use all building/pushing image targets
+IMG ?= quay.io/kiegroup/kogito-cloud-operator:$(VERSION)
```
The purpose of BUNDLE_IMG and IMG are different I'll change the value of BUNDLE_IMG, this BUNDLE_IMG only contains our CRD and CSV while the IMG is for our operator container image

Kaitou786 replied:
Changed the name of BUNDLE_IMG

Kaitou786 commented on .github/workflows/Kogito-Operator-Unit-tests.yaml:
```diff
@@ -29,26 +31,36 @@ jobs:
+      - name: Run unit tests
         run: |
-          go test ./cmd/... -count=1 -p=1 -coverprofile=cmd-coverage.txt -covermode=atomic
+          ./hack/go-test.sh
```
Because we are running `lint` and `fmt` as part of our `make test`. Since we already have a different GH action to check those I would just run the unit tests here. :)

radtriste replied:
ok. as we have a Makefile that will call different scripts, I am just worried about bypassing the Makefile and maybe later changes in the `test` command that would not be taken into account the GHAction

Kaitou786 replied:
Yeah, that is true, although we had this same structure earlier as well, since in github actions we only want to run unit tests but in `make test` we do other things like check lint and format the code as well.

Kaitou786 commented on PROJECT:
```diff
@@ -0,0 +1,20 @@
+domain: kiegroup.org
```
This is the group which is appended to our CRDs right so right now, so right now we have CRD as `kogitoinfras.app.kiegroup.org` of we change it becomes `kogitoinfras.app.kie.org`, I think we can think about changing it later but not in this same PR, personally I am not sure if changing the group in a middle version is a good idea but this can be discussed :)

radtriste replied:
fair enough :)

Kaitou786 commented on bundle.Dockerfile:
```diff
@@ -0,0 +1,15 @@
+FROM scratch
```
I would rather not, this is the file generated by the `make bundle` and will be autogenrated if not found whenever we run `make bundle` this will be genrated in the root of our dir

radtriste replied:
oh ok ...

Kaitou786 commented on examples/data-index-mongodb.yaml:
```diff
@@ -32,10 +32,10 @@ metadata:
   serviceType: DataIndex
-  # environment variables to set in the runtime container. Example: JAVA_OPTIONS: "-Dquarkus.log.level=DEBUG"
-  #env:
+    # environment variables to set in the runtime container. Example: JAVA_OPTIONS: "-Dquarkus.log.level=DEBUG"
```
Sorry, rebase problem, I'll change it back :)

Kaitou786 replied:
should be okay now :)

Kaitou786 commented on hack/export-version.sh:
```diff
@@ -14,6 +14,13 @@
+echo "Operator version is ${OP_VERSION}"
+
+### Fetching the latest released kogito-cloud-operator on the OperatorHub
+rm -rf /tmp/operators/
```
Yeah, good idea will change it :)

radtriste commented on hack/go-test.sh:
```diff
@@ -13,23 +13,16 @@
+
+testdir=$(pwd)/${ENVTEST_ASSETS_DIR}
+mkdir -p ${testdir}
+test -f  ${testdir}/setup-envtest.sh || curl -sSLo testbin/setup-envtest.sh https://raw.githubusercontent.com/kubernetes-sigs/controller-runtime/v0.6.3/hack/setup-envtest.sh
```
```suggestion
test -f  ${testdir}/setup-envtest.sh || curl -sSLo ${testdir}/setup-envtest.sh https://raw.githubusercontent.com/kubernetes-sigs/controller-runtime/v0.6.3/hack/setup-envtest.sh
```

radtriste reviewed (approved):
one minor thing. Otherwise lgtm. Nice work !

ricardozanini commented:
Hey @sutaakar I'm assuming you're working on BDD + verifications for this one, right? Do you think we will be able to manage before the holidays?

sutaakar commented:
@ricardozanini right, I plan to have BDD tests running back today

sutaakar commented:
BDD test adjustment prepared - https://github.com/Kaitou786/kogito-cloud-operator/pull/46

sutaakar reviewed (approved):
Looks good, seems to work (based on a few local BDD test execution and brief experiments).
I would have some minor comments about logging, but that can be addressed separately.

ricardozanini commented:
@Kaitou786 we can wrap up this PR on Monday.
@spolti do you mind taking a look? If you have any comments, Tarun can address them on Monday, so we close in time for 1.1.0.

spolti commented:
@Kaitou786 please rebase.

ricardozanini commented:
/jenkins test

Kaitou786 commented:
/jenkins test

Kaitou786 commented:
I just changed the image on the configuration of jenkins with updated operator-sdk 1.2, And reverted it back so this updated image is only used for this PR :)
That's why aborted the previous job and restarted it

ricardozanini commented:
@Kaitou786 can't you re-run the job using this comment section? Now it will keep this ugly red mark in the PR. :)

Kaitou786 commented:
Oh the status will be updated here only, that's what I did, the error was coming from `buildah` it is now `bud` instead of `build` when using `buildah`
/jenkins test

Kaitou786 commented:
Seeing the error the only test failing are of Infinispan, Grafana and Prometheus

Kaitou786 commented:
/jenkins test

Kaitou786 commented:
/jenkins test

Kaitou786 commented:
@ricardozanini can be merged now =D

spolti commented on README.md:
```diff
@@ -96,7 +96,7 @@ After you update the source with your new proposed feature or bug fix, open a [p
 - The PR solves only the problem described in the JIRA.
 - You have written unit tests for the particular fix or feature.
-- You ran `make test` before submitting the PR and everything is working accordingly.
+- You ran `make vet` and `make test` before submitting the PR and everything is working accordingly.
```
can't we have make vet being executed as part of make test?

Kaitou786 replied:
We can, I didn't change this, I mean the purpose of make vet and make test, it was  this way earlier also

radtriste replied:
Vet and test are different things. We could have a third command which would do both. But I would say this is for another issue ;)

spolti replied:
sounds good.

spolti commented:
@Kaitou786 really nice work.
One thing, for people that already have the code base locally, besides the operator-sdk update and rebase, is there anything else needed?

Kaitou786 commented:
> @Kaitou786 really nice work.
> One thing, for people that already have the code base locally, besides the operator-sdk update and rebase, is there anything else needed?

No not really everything else will work :)

Kevin-Mok commented:
When I try to run `make vet`, I'm getting the following error:
```
/home/kevin/go/bin/controller-gen object:headerFile="hack/boilerplate.go.txt" paths="./..."
./hack/openapi.sh
Generating openapi files
API rule violation: names_match,github.com/kiegroup/kogito-cloud-operator/api/v1beta1,KogitoInfraStatus,Volume
/home/kevin/go/bin/controller-gen "crd:trivialVersions=true" rbac:roleName=manager-role webhook paths="./..." output:crd:artifacts:config=config/crd/bases
cd config/manager && /usr/bin/kustomize edit set image controller=quay.io/kiegroup/kogito-cloud-operator:2.0.0-snapshot
/usr/bin/kustomize build config/default > kogito-operator.yaml
panic: interface conversion: interface {} is string, not map[string]interface {}

goroutine 1 [running]:
sigs.k8s.io/kustomize/api/resmap.getNamespacesForRoleBinding(0xc0004f34a0, 0x19)
	sigs.k8s.io/kustomize/api@v0.7.0/resmap/reswrangler.go:417 +0x352
sigs.k8s.io/kustomize/api/resmap.(*resWrangler).SubsetThatCouldBeReferencedByResource(0xc0004e3940, 0xc0004f34a0, 0xb, 0xc001f98701)
	sigs.k8s.io/kustomize/api@v0.7.0/resmap/reswrangler.go:384 +0xdb
sigs.k8s.io/kustomize/api/internal/accumulator.(*nameReferenceTransformer).Transform(0xc0008a6320, 0x5629e987bf80, 0xc0004e3940, 0xc0004e3940, 0x0)
	sigs.k8s.io/kustomize/api@v0.7.0/internal/accumulator/namereferencetransformer.go:85 +0x531
sigs.k8s.io/kustomize/api/internal/accumulator.(*ResAccumulator).Transform(...)
	sigs.k8s.io/kustomize/api@v0.7.0/internal/accumulator/resaccumulator.go:138
sigs.k8s.io/kustomize/api/internal/accumulator.(*ResAccumulator).FixBackReferences(0xc0004e3920, 0xc0004e3920, 0x0)
	sigs.k8s.io/kustomize/api@v0.7.0/internal/accumulator/resaccumulator.go:164 +0xbe
sigs.k8s.io/kustomize/api/internal/target.(*KustTarget).makeCustomizedResMap(0xc00058c640, 0x0, 0x0, 0x5629e98761c0, 0x5629ea2ae820)
	sigs.k8s.io/kustomize/api@v0.7.0/internal/target/kusttarget.go:125 +0x7d
sigs.k8s.io/kustomize/api/internal/target.(*KustTarget).MakeCustomizedResMap(...)
	sigs.k8s.io/kustomize/api@v0.7.0/internal/target/kusttarget.go:106
sigs.k8s.io/kustomize/api/krusty.(*Kustomizer).Run(0xc0012c3d20, 0x7fff12c65ffd, 0xe, 0x0, 0x0, 0x0, 0x0)
	sigs.k8s.io/kustomize/api@v0.7.0/krusty/kustomizer.go:77 +0x29b
sigs.k8s.io/kustomize/kustomize/v3/internal/commands/build.(*Options).RunBuild(0xc0004c20e0, 0x5629e9828d40, 0xc000010018, 0x1, 0x0)
	sigs.k8s.io/kustomize/kustomize/v3/internal/commands/build/build.go:148 +0xb7
sigs.k8s.io/kustomize/kustomize/v3/internal/commands/build.NewCmdBuild.func1(0xc0002dfb80, 0xc0004b5880, 0x1, 0x1, 0x0, 0x0)
	sigs.k8s.io/kustomize/kustomize/v3/internal/commands/build/build.go:70 +0xad
github.com/spf13/cobra.(*Command).execute(0xc0002dfb80, 0xc0004b5850, 0x1, 0x1, 0xc0002dfb80, 0xc0004b5850)
	github.com/spf13/cobra@v1.0.0/command.go:842 +0x47c
github.com/spf13/cobra.(*Command).ExecuteC(0xc0002df600, 0x0, 0xffffffff, 0xc00009e058)
	github.com/spf13/cobra@v1.0.0/command.go:950 +0x375
github.com/spf13/cobra.(*Command).Execute(...)
	github.com/spf13/cobra@v1.0.0/command.go:887
main.main()
	sigs.k8s.io/kustomize/kustomize/v3/main.go:19 +0x2a
make: *** [Makefile:138: generate-installer] Error 2
```
The API rule violation I fixed in my PR for KOGITO-3777 and [is discussed there](https://github.com/kiegroup/kogito-cloud-operator/pull/700#discussion_r545955366). I'm not sure what's going on with the interface conversion error though?

ricardozanini commented:
@Kaitou786 since you did this migration, can you help @Kevin-Mok ?

Kaitou786 commented:
@Kevin-Mok the error can be related to missed matched version kustomize tool, can you try running:
`rm $(which kustomize)`
and then run `make vet` again?

Kevin-Mok commented:
That fixed it. Thanks!
